File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_SpriteChar.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_SpriteChar
*/
public class t_SpriteChar : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public ushort	id;
	public Vector4	uv;
	public sbyte	page;
	public sbyte	xOffset;
	public sbyte	yOffset;
	public sbyte	xAdvance;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_SpriteChar() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		id = 0;
		page = 0;
		xOffset = 0;
		yOffset = 0;
		xAdvance = 0;
	}
	public bool read(CReadVariable cVariable) {
		try {
			id = cVariable.getU16();
			uv = cVariable.getVector4();
			page = cVariable.getS8();
			xOffset = cVariable.getS8();
			yOffset = cVariable.getS8();
			xAdvance = cVariable.getS8();
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(id);
		cVariable.put(uv);
		cVariable.put(page);
		cVariable.put(xOffset);
		cVariable.put(yOffset);
		cVariable.put(xAdvance);
		return true;
	}
};

File: Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_SpriteChar.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
#/==========================================================================
#*!
#	@brief	t_SpriteChar
#/
class t_SpriteChar:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	id = 0	#U16
	uv = Vector4()
	page = 0	#S8
	xOffset = 0	#S8
	yOffset = 0	#S8
	xAdvance = 0	#S8
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.id = 0
		self.uv = Vector4()
		self.page = 0
		self.xOffset = 0
		self.yOffset = 0
		self.xAdvance = 0

	def read(self,cVariable):
		self.id = cVariable.getU16()
		self.uv = cVariable.getVector4()
		self.page = cVariable.getS8()
		self.xOffset = cVariable.getS8()
		self.yOffset = cVariable.getS8()
		self.xAdvance = cVariable.getS8()
		return True

	def write(self,cVariable):
		cVariable.putU16(self.id)
		cVariable.putVector4(self.uv)
		cVariable.putS8(self.page)
		cVariable.putS8(self.xOffset)
		cVariable.putS8(self.yOffset)
		cVariable.putS8(self.xAdvance)
		return True

